### PR TITLE
Sync with master

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: resources/templates/ansic_cmake_iotcsaskey/Readme.md [hunk 1]
```diff
@@ -16,50 +17,45 @@ For more details about setting up your development environment for compiling the
...
-        ```bash
-        \\{PROJECT_NAME}\\Release\\{PROJECT_NAME}.exe
-        ```
+    * Retrieve DPS connection infomation from Azure IoT Central, including **DPS ID Scope**, **DPS Symmetric Key**, **Device ID**, which will be pass the as paramerters of the device app executable. Please refer to [this document](https://docs.microsoft.com/en-us/azure/iot-central/concepts-connectivity) for more details.
 
-    * If you want to pass the DPS info as command line paramers.
-        ```bash
-        \\{PROJECT_NAME}\\Release\\{PROJECT_NAME}.exe [DPS ID Scope] [DPS symmetric key] [device ID]
-        ```
+1. Once the build has succeeded, you can test it by specifying the the DPS info (**DPS ID Scope**, **DPS Symmetric Key**, **Device Id**) as its parameter.
```
Comment (eriolchan) [inline resources/templates/ansic_cmake_iotcsaskey/Readme.md]: duplicate "the" in the front of DPS

File: src/Models/ArduinoDeviceBase.ts [hunk 2]
```diff
@@ -166,17 +165,15 @@ export abstract class ArduinoDeviceBase implements Device {
...
 
-  async generateCppPropertiesFile(
-      type: ScaffoldType, board: Board,
-      fileInfo: TemplateFileInfo): Promise<void> {
-    const targetFolder = path.join(this.deviceFolder, fileInfo.targetPath);
-    if (!await FileUtility.directoryExists(type, targetFolder)) {
-      await FileUtility.mkdirRecursively(type, targetFolder);
+  async generateCppPropertiesFile(type: ScaffoldType, board: Board):
+      Promise<void> {
+    if (!await FileUtility.directoryExists(type, this.vscodeFolderPath)) {
```
Comment (eriolchan) [inline src/Models/ArduinoDeviceBase.ts]: can we merge them to ensureDirectory()?

Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Yes. Consider fixing this in next iteration.

Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Yes. Consider fixing this in next iteration.

File: resources/templates/ansic_cmake_iotcsaskey/Readme.md (hunk 1 above)
Comment (dilin-MS) [reply resources/templates/ansic_cmake_iotcsaskey/Readme.md]: Nice catch. @dooriya will fix it.

**Pull request merged**